core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -520,8 +520,18 @@ private[spark] class BlockManagerInfo(
         blockStatus = BlockStatus(storageLevel, memSize = memSize, diskSize = 0)
         _blocks.put(blockId, blockStatus)
         _remainingMem -= memSize
-        logInfo("Added %s in memory on %s (size: %s, free: %s)".format(
-          blockId, blockManagerId.hostPort, Utils.bytesToString(memSize),
+        val originalMemSize = if (_blocks.containsKey(blockId)) {
+          _blocks.get(blockId).memSize
+        }  else {
+          0
+        }
+        val (addedOrRemoved, size) = if (memSize >= originalMemSize) {
+          ("Added", memSize - originalMemSize)
+        } else {
+          ("Removed", originalMemSize - memSize)
+        }
+        logInfo("%s %s in memory on %s (size: %s, free: %s)".format(
+          addedOrRemoved, blockId, blockManagerId.hostPort, Utils.bytesToString(size),
           Utils.bytesToString(_remainingMem)))
       }
       if (storageLevel.useDisk) {
```

Comment:
Ok, I have modified the disk case, and add a "update" style log info, any other suggestions? thank you